### ISSUE TITLE
Decimal.TryParse overload is missing NumberStyles argument

### ISSUE THREAD

**Issue by Porl91** [bridgedotnet/Bridge#1947]
Currently the only type of number allowed by any decimal.TryParse overloads is a string containing a standard floating point number, as there are no overloads that accept the NumberStyles enum.

Example of desired behaviour:
https://dotnetfiddle.net/vLS2YG

What's currently allowed by the decimal.TryParse overloads:
http://deck.net/91d90aef2e1330a98dd7714babaa0fe9

In this example I'm trying to parse currency, although TryParse fails to parse if I just pass in the currency string.

This also seems to be the case for single.TryParse / double.TryParse.

**Comment by kcudnik** [comment]
Seems like not easy task [impl](https://referencesource.microsoft.com/#mscorlib/system/number.cs,04291cc3a0b10032)

**Comment by samuelGrahame** [comment]
Yeah as @kcudnik points out it is not an easy task, instead of puting all that code in to support it.

This is what I would do in the mean time: (Make a Currency Class)

http://deck.net/24733f4c471f3331ebcc12d907bf0f42
